src/pool/srv_pool.c

```diff
@@ -2338,6 +2347,31 @@ ds_pool_connect_handler(crt_rpc_t *rpc)
 	}
 	D_ASSERT(prop != NULL);
 
+	global_ver_entry = daos_prop_entry_get(prop, DAOS_PROP_PO_GLOBAL_VERSION);
+	D_ASSERT(global_ver_entry != NULL);
+	global_ver = global_ver_entry->dpe_val;
+	/*
+	 * Reject pool connection if client is unable to access newer pool.
+	 */
+	if (handler_version > 0) {
+		struct pool_connect_1_in *in1 = (struct pool_connect_1_in *)in;
+
+		if (in1->pci_pool_version <= DAOS_PROP_PO_GLOBAL_VERSION) {
+			if (global_ver > pool_max_supported_version[in1->pci_pool_version]) {
+				D_ERROR(DF_UUID": cannot connect, try to upgrade client firstly?\n",
+					DP_UUID(in->pci_op.pi_uuid));
+				D_GOTO(out_map_version, rc = -DER_NOTSUPPORTED);
+			}
+		}
+	} else {
+		/* Assuming 2.0 client */
+		if (global_ver > 1) {
```

Comment:
I think yes, with EC rotation and corresponding upgrade changes, DAOS_POOL_GLOBAL_VERSION will be bumped to 2.

Reply:
So at this point this branch will not be taken in the code since it will need to be tested with another patch that will increment DAOS_POOL_GLOBAL_VERSION?

Reply:
Yup